model/task_queue_service.go

```diff
@@ -73,34 +76,59 @@ func (s *taskDispatchService) Refresh(distro string) error {
 	return nil
 }
 
-func (s *taskDispatchService) ensureQueue(distro string) (CachedDispatcher, error) {
+func (s *taskDispatchService) ensureQueue(distroID string) (CachedDispatcher, error) {
 	// If there is a "distro": *basicCachedDispatcherImpl in the cachedDispatchers map, return that.
 	// Otherwise, get the "distro"'s taskQueue from the database; seed its cachedDispatcher; put that in the map and return it.
 	s.mu.RLock()
 
-	distroDispatchService, ok := s.cachedDispatchers[distro]
+	distroDispatchService, ok := s.cachedDispatchers[distroID]
 	s.mu.RUnlock()
 	if ok {
 		return distroDispatchService, nil
 	}
 
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	distroDispatchService, ok = s.cachedDispatchers[distro]
+	distroDispatchService, ok = s.cachedDispatchers[distroID]
 	if ok {
 		return distroDispatchService, nil
 	}
 
-	taskQueue, err := FindDistroTaskQueue(distro)
+	taskQueue, err := FindDistroTaskQueue(distroID)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
 	taskQueueItems := taskQueue.Queue
-	distroDispatchService = newDistroTaskDispatchService(distro, taskQueueItems, s.ttl)
-	s.cachedDispatchers[distro] = distroDispatchService
 
-	return distroDispatchService, nil
+	// TODO : do we need to make a call to the database to get the distro's PlannerSettings (which we probably also need to resolve a la GetResolvedPlannerSettings())
+	// Should we be using the distro.PlannerSettings.Version field to signify which DispatchService to use, or has the distro.PlannerSettings.Version field become overloaded?
+	// Do we need to independently toggle the task planning (PlannerSettings) and task dispatcher (DispatcherSettings) options for a given distro?
+	// Just going with this stop-gap for now to get things working...
```

Comment:
I think it's fine to make these calls, because they're cheap (idhack queries...) and because of the caching we're not doing them that frequently. If the caching doesn't protect us from doing this on every request (it's ok everytime we reload the queue.) then I think its fine.

Reply:
As discussed in person: the `Task Planner` and `Task Dispatcher` versions will always be in lock step, so the `PlannerSettings.Version` field is sufficient.